### PR TITLE
Update user

### PR DESCRIPTION
**Pull request by zygimantasjacikevicius**

### CODE REVIEW

File: app/Http/Controllers/UserController.php [hunk 1]
```diff
@@ -21,11 +23,13 @@ class UserController extends Controller
     protected UserService $userService;
     protected ResetPasswordService $resetPasswordService;
     protected ResetPasswordService $newPasswordService;
+    protected UpdateUserService $updateUserService;
 
-    public function __construct(UserService $userService, ResetPasswordService $resetPasswordService)
+    public function __construct(UserService $userService, ResetPasswordService $resetPasswordService, UpdateUserService $updateUserService)
```
Comment (Pavel-Tsymbal) [inline app/Http/Controllers/UserController.php]: use userService instead of UpdateUserService

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -21,11 +23,13 @@ class UserController extends Controller
...
     {
         $this->userService = $userService;
         $this->resetPasswordService = $resetPasswordService;
+        $this->updateUserService = $updateUserService;
```
Comment (Pavel-Tsymbal) [inline app/Http/Controllers/UserController.php]: remove this line

File: app/Http/Controllers/UserController.php [hunk 3]
```diff
@@ -60,4 +64,10 @@ public function newPassword(NewPassword $request)
         $this->resetPasswordService->newPassword($request->validated());
         return response()->json(['success' => 'Your password has been changed'], 200);
     }
+
+    public function update(UpdateUser $request)
+    {
+        $this->updateUserService->updateUser($request->validated());
```
Comment (Pavel-Tsymbal) [inline app/Http/Controllers/UserController.php]: $this->userService.....

File: app/Services/UpdateUserService.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace App\Services;
+
+use Illuminate\Support\Facades\Auth;
+
+class UpdateUserService
+{
+    public function updateUser(array $userData)
+    {
+        $user = Auth::user();
```
Comment (Pavel-Tsymbal) [inline app/Services/UpdateUserService.php]: remove this line. 
pass user as an argument: public function updateUser(User $user, array $userData)...

File: app/Services/UpdateUserService.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+        if (!empty($userData['new_password'])) {
+            $user->password = $userData['new_password'];
```
Comment (Pavel-Tsymbal) [inline app/Services/UpdateUserService.php]: make sure that password will be bcrypted before user save

File: app/Services/UpdateUserService.php [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        }
+        if (!empty($userData['email'])) {
+            $user->email = $userData['email'];
+        }
+
+        if ($user->can('update', $user)) {
```
Comment (Pavel-Tsymbal) [inline app/Services/UpdateUserService.php]: move this check to your request authorize method

return $user->can('update', $this->user)

File: app/Services/UpdateUserService.php (hunk 5 above)
Comment (zygimantasjacikevicius) [reply app/Services/UpdateUserService.php]: I have bcrypt in the User model, do I need to bcrypt again?

**Pull request merged**